src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -364,6 +463,22 @@ protected static boolean regexListMatch(List<String> regexList, String input) {
       return false;
     }
 
+    /**
+     * Check if a metric is matched in `aws_tag_select`
+     */
+    private boolean metricIsInAwsTagSelect(MetricRule rule, List<String> tagBasedResourceIds, Metric metric) {
+      for (Dimension dimension : metric.getDimensions()) {
+        String dimensionName = dimension.getName();
+        String dimensionValue = dimension.getValue();
+        if (dimensionName.equals(rule.awsTagSelect.resourceIdDimension)) {
```

Comment:
do the opposite for null checking ?

Reply:
```suggestion
        if (rule.awsTagSelect.resourceIdDimension.equals(dimensionName)) {
```

Reply:
@vincentcreusotfx  Not sure it was needed, but why not!